exp/api/v1beta1/azuremachinepool_types.go

```diff
@@ -135,6 +135,16 @@ type (
 		// +optional
 		Identity infrav1.VMIdentity `json:"identity,omitempty"`
 
+		// Scope is the scope that the role assignment should be created at. If left empty, will default to the
+		// subscription scope.
+		// +optional
+		Scope string `json:"scope,omitempty"`
```

Comment:
Should this Field `Scope` and the other new one `RoleDefinitionID` have a name that includes somehow their relationship to the `SystemAssigned` identity  ? 

something like `SystemAssignedIdentityScope` and `SystemAssignedIdentityRoleDefinitionID` would probably be too verbose ... but something that helps understand what they refer to ?

Reply:
I do agree that it would be too verbose. Maybe I can add a more descriptive comment. I am also planning on adding a check in the validating webhook to ensure the `identity` field is being set if `Scope` and/or `RoleDefinitionID` is specified.

@CecileRobertMichon any thoughts on how to name this better?

Reply:
Needing to prefix a bunch of fields with the same value is usually a sign that the API isn't properly structured. It does seem like the right thing to do here would be to re-organize the API to something like:

```
spec
...
identity:
  type: SystemAssigned
  scope: ...
  role: ...
  roleAssignmentName: ...
``` 

instead of having a bunch of identity-related fields in top level spec. Unfortunately that would be a breaking API change so we would only be able to do this if we bump the API version. Would be good to open an issue in any case so we can flag it as something we'd want to do in vnext.

For now maybe at least combining these two new fields into a struct like this would be a good first step:

```
systemAssignedIdentityRole:
  definitionID: // RoleDefinitionID
  scope: // Scope
  name: // RoleAssignmentName (note: this would be breaking, could omit for now or make it backwards compatible)
```